frontend/src/types/promotion.ts

```diff
@@ -0,0 +1,82 @@
+export interface Promotion {
+  id: string;
+  category: string;
+  cuisine: string;
+  dateAdded: string;
+  description: string;
+  discount: Discount;
+  expirationDate: string;
+  image: PromotionImage;
+  liked: boolean;
+  name: string;
+  restaurantName: string;
+  user: User;
+}
+
+export interface User { 
+  id: string;
+  email: string;
+  firstName: string;
+  lastName: string;
+  password: string;
+  username: string;
+}
+
+export interface Discount {
+  id: string;
+  discountValue: number;
+  type: string;
+}
+
+export interface PromotionImage {
+  src: string;
+}
+
+export enum Category {
```

Comment:
I think there may be some miscommunication on the meaning of "Category". Backend and design seem to have this as promotion category (ie. bogo, happy hour, etc), but I agree with you that it makes more sense that category refers to restaurant category (especially when "Discount Type" is itself a thing)

Reply:
Ah I see okay so this is a new DB schema, this makes more sense I agree.

So just to double-check, a promotions category will go more specific into the cuisine type. And then the `Discount` entity will tell us the discount type and discount category? Discount type is for specific numeric units like percentage, dollar amount etc. But discount category will still exist to be Bogo, happy hour?

One thing I'm thinking about is now the complexity of preventing users from choosing a category that does not belong in the cuisine type. For example, a user could create a promotion with a category "curry", but then call that cuisine type "Mexican"

Reply:
This will take some time to refactor in the BE too unfortunately

Reply:
I'm not sure that a discount category is necessary to be honest. I think the current schema should be good as long as the "category" field is understood to be restaurant category. Also, I think examples of restaurant category would be like the ones below (ie. fast food, fast casual, casual dining, contemporary casual, cafe)

Reply:
But then how would a user filter for promotions that are specifically happy hour, BOGO, seasonal etc

Reply:
In terms of restaurant categories, I can also imagine this being supported by our search query functionality. The user can just search up fast food, and we can search inside the promotion name and description for data like this

Reply:
My thinking was that users wouldn't see much use in filtering by discount category, but I could be wrong. It's probably best to keep discount category though and add it to the discount table if it's not too much work @ericrkuo

Reply:
@richardyjtian @ericrkuo Ya, I think it got lost in translation somewhere... So I think there are two types of categories: Restaurant Categories ("bubble tea", "dessert", "fast food") and Promotion Categories (BOGO, Happy Hour). I had a discussion with Wendy and Elina last week about it and they gave me these categories, but then they didn't account for the promotion categories. Now I'm looking at the hi-fi designs and they're both missing so yeah I'll need to clear this up!

Reply:
<img width="320" alt="Screen Shot 2020-11-18 at 9 07 02 PM" src="https://user-images.githubusercontent.com/44531733/99624027-3f859980-29e2-11eb-88b7-3c3ffee12781.png">
<img width="537" alt="Screen Shot 2020-11-18 at 9 07 27 PM" src="https://user-images.githubusercontent.com/44531733/99624050-4dd3b580-29e2-11eb-8ed7-e529bdd4e268.png">

@richardyjtian @ericrkuo What do you guys think about this?

Reply:
> My thinking was that users wouldn't see much use in filtering by discount category, but I could be wrong. It's probably best to keep discount category though and add it to the discount table if it's not too much work @ericrkuo

Ah yea I see what you mean Richard that makes sense. Like something more useful would be to search a quantifiable filter, like promotions with greater than 10$ off, or greater than 50% off. rather than to filter by %, $, etc.

@andrewzulaybar Elina mentioned to me that the original reasoning for not including restaurant categories was because this could be supported by the search functionality. But then one could also argue, Cuisine Type could also be supported by the search functionality.

Reply:
Oops i forgot to refresh before sending my message, but yes I like Elina's idea. I also like richards idea with the discount type

Reply:
actually now i am confused, what happened to Promotion Category LOL

Reply:
> Ah yea I see what you mean Richard that makes sense. Like something more useful would be to search a quantifiable filter, like promotions with greater than 10$ off, or greater than 50% off. rather than to filter by %, $, etc. @andrewzulaybar would you mind creating an issue for this if you're oaky with the idea?

By search by a quantifiable filter do you mean using the search bar?

Reply:
> actually now i am confused, what happened to Promotion Category LOL is there a filter for that still?

Ya I'm a bit confused too haha let me check in with Elina

Reply:
<img width="262" alt="Screen Shot 2020-11-18 at 9 20 06 PM" src="https://user-images.githubusercontent.com/44531733/99624984-136b1800-29e4-11eb-8711-6410286315d6.png">

@ericrkuo I think they're here now? They've combined it with the "Discount Type" but Elina's asking me if "Promotion Type" would be a better name for this dropdown

Reply:
Hmmm
looking at the figma it seems Promotion Category and Discount Type are merged together. That could work imo
![image](https://user-images.githubusercontent.com/49849754/99625081-444b4d00-29e4-11eb-95f4-2e70dd348185.png)

Reply:
Oops didnt refresh again sry